LAVA_setup_guide.md

```diff
@@ -0,0 +1,192 @@
+    * You can just copy the whole directory https://github.com/ARMmbed/mbed-os-lava-scripts/tree/master/lava-server to /etc/, otherwise follow steps below
+    * must have extension jinja2: `mbed.jinja2`
+    * the device should inherit from a device type in /usr/share/lava-server/device-types, to inherit from docker add line: `{% extends "docker.jinja2" %}`
+    * Must pass in extra parameters to docker: `{% set docker_extra_arguments = ["-v /dev:/dev --group-add=dialout --device-cgroup-rule 'a 166:* rwm'"] %}`
+- `sudo service lava-server-gunicorn restart`
+
+### Worker setup
+
+#### Install dependencies
+
+`sudo apt install docker.io git`
+Install docker following official [docker documentation](https://docs.docker.com/engine/install/debian/).
+`sudo apt install bluez bluetooth`
+`sudo killall -9 bluetoothd`
+    * daemon cannot be running when docker starts
+    * you can change AutoEnable=true to false in /etc/bluetooth/main.conf but it still needs to start at least once
+
+#### Install worker
+
+`sudo apt install lava-dispatcher`
+`sudo vim /etc/lava-dispatcher/lava-worker`
+    * set WORKER_NAME and URL
+    * set the token to the string you get from adding a worker in the server
+
+#### Device access
+
+`sudo usermod -aG dialout $(whoami)`
+`git clone https://github.com/pyocd/pyOCD.git`
+`sudo cp pyOCD/udev/*.rules /etc/udev/rules.d`
+
+`sudo nano /etc/dbus-1/system.d/bluetooth.conf`
+    * copy the bluetooth policy from root to lava-admin
+
+#### Create workspace
+
+Some work is done locally on the worker machine. This is the scratch space used for these operations to speed them
+up and avoid having to recreate the same environment every time.
+
+`sudo mkdir -p /opt/lava-worker`
+`sudo chmod 777 /opt/lava-worker`
+`cd /opt/lava-worker`
+`git clone https://github.com/ARMmbed/mbed-os-lava-scripts lava-scripts`
+
+## Running LAVA server docker compose
+
+Instead of installing your lava server you may use docker compose provided by us.
+
+- `git clone https://github.com/ARMmbed/mbed-os-lava-docker-compose`
+-  Create a certificate and key and replace the ones in apache2/cert (you can try the free cert from
+   [zerossl](https://app.zerossl.com/certificate/new))
+- `docker-compose build lava-server`
+- `docker-compose up lava-server`
+
+Website is available on localhost, username and password are `lava-admin`. Change the password after logging in.
+
+## Running a premade SD card image with the worker on your RPi
+
+Download the SD card image [MISSING LINK](http://) and use [Raspberry PI imager](https://www.raspberrypi.org/software)
+to copy to the SD card.
+
+You will need to edit the config file to give the worker its unique name and password. If you're using a different
+server you will also need to change the server URL.
+
+`sudo vim /etc/lava-dispatcher/lava-worker`
+    * set WORKER_NAME (and URL if needed)
+    * set the token to the string you get from adding a worker in the server
+
```

Comment:
Maybe its worth to mention that this whole setup is fixed to 2021.03.post1 version of lava server/worker.

Reply:
currently it is

Reply:
Version provided from the package repository is 2021.05

Reply:
so what?

Reply:
I don't know what you're both driving at, please make an explicit suggestion

Reply:
The manual setup is not tied to any one version. The docker compose you don't need to know what version it is - you will use whatever version is inside it which might change in the future and has no bearing on this guide.

Reply:
With the current setup, any `apt-get upgrade` will break the system as it will update the lava version on the worker. 
I think it is worth mentioning it in the guide.

Reply:
It is mentioned in the guide